src/app/avatar/heygen/index.tsx

```diff
@@ -0,0 +1,174 @@
+'use client'
+import React, { useState, useEffect, useRef, useCallback } from 'react';
+import { handleICE, newSession, repeat, startSession, stopSession } from './helper';
+
+// Interface for session info
+interface SessionInfo {
+  session_id: string;
+  sdp: RTCSessionDescriptionInit;
+  ice_servers2: RTCIceServer[];
+}
+
+
+export function HeyGenPage({text}:{text: string}) {
+  // State variables
+  const [status, setStatus] = useState<string>('');
+  const [sessionInfo, setSessionInfo] = useState<SessionInfo | null>(null);
+  const [talkingToAI, setTalkingToAI] = useState(false);
+  const peerConnectionRef = useRef<RTCPeerConnection | null>(null);
```

Comment:
Might be worth breaking this out into its own class - I bet the other implementations will need something similar

Reply:
I think I'll break it out after I've connected a few more providers, so its clear to me where the overlap is.